examples/step-55/doc/intro.dox

```diff
@@ -0,0 +1,141 @@
+setup times, solve times, and number of iterations. We also compute
+errors to make sure that what we have implemented is correct.
+
+Let $u \in H_0^1 = \{ u \in H^1(\Omega), u|_{\partial \Omega} = 0 \}$
+and $p \in L_*^2 = \{ p_f \in L^2(\Omega), \int_\Omega p_f = 0
+\}$. The Stokes equations read as follows in non-dimensionalized form:
+
+@f{eqnarray*} - 2 \text{div} \frac {1}{2} \left[ (\nabla \textbf{u}) +
+(\nabla \textbf{u})^T\right] + \nabla p & =& f \\ - \nabla \cdot u &=&
+0 @f}
+
+Note that we are using the deformation tensor instead of $\Delta u$ (a
+detailed desription of the difference between the two can be found in
+Step-22, but in summary, the deformation tensor is more physical as
+well as more expensive).
+
+<h3> Linear solver and preconditioning issues </h3> The weak form of
+the discrete equations naturally leads to the following linear system
+for the nodal values of the velocity and pressure fields:
+@f{eqnarray*} 
+\left(\begin{array}{cc} A & B^T \\ B & 0
+\end{array}\right) \left(\begin{array}{c} U \\ P \end{array}\right) =
+\left(\begin{array}{c} F \\ 0 \end{array}\right), @f}
+
+Our goal is to compare several solver approaches.  In contrast to the
+way in which step-22 solves the Stokes equation, we instead attack the
+block system at once using a direct solver or FMGRES with an efficient
+preconditioner. The idea is as follows: if we find a block
+preconditioner $P$ such that the matrix @f{eqnarray*}
+\left(\begin{array}{cc} A & B^T \\ B & 0 \end{array}\right) P^{-1} @f}
+
+is simple, then an iterative solver with that preconditioner will
+converge in a few iterations. Notice that we are doing right
+preconditioning for this.  Using the Schur complement $S=BA^{-1}B^T$,
+we find that
+
+@f{eqnarray*}
+P^{-1} = \left(\begin{array}{cc} \hat{A} & B^T \\ 0 &
+ \hat{S} \end{array}\right)^{-1} @f}
+
+is a good choice. It is important to note that
+@f{eqnarray*} 
+P =
+\left(\begin{array}{cc} A^{-1} & 0 \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & B^T \\ 0 & -I \end{array}\right)
+\left(\begin{array}{cc} I & 0 \\ 0 & S^{-1} \end{array}\right).  @f}
+
+Since $P$ is aimed to be a preconditioner only, we shall use
+approximations to the inverse of the Schur complement $S$ and the
+matrix $A$.  Therefore, in the above equations, $-M_p=\hat{S} \approx
+S$, where $M_p$ is the pressure mass matrix and is solved by using CG
++ ILU, and $\hat{A}$ is an approximation of $A$ obtained by one of
+multiple methods: CG + ILU, just using ILU, CG + GMG (Geometric
+Multigrid as described in Step-16), or just performing a few V-cycles
+of GMG. The inclusion of CG is more expensive, in general.
+
+As a comparison, instead of FGMRES, we also use the direct solver
+UMFPACK to compare our results to.  If you want to use UMFPACK as a
+solver, it is important to note that since you have a singular system
+(since the integral of mean pressure being equal to zero not
+implemented), we set the first pressure node equal to zero since the
```

Comment:
this reads a bit strange. maybe say "it is important to set the first pressure node equal to zero to avoid the system being singular (recall that the Stokes equation itself only determines the pressure up to a constant when using only Dirichlet boundary conditions). If we do not do this, then the direct solver will produce an error message whereas the iterative solvers quietly solve it anyway."